src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -705,9 +718,21 @@ bool RTPSParticipantImpl::createReader(
         return false;
     }
 
+    // Check if also support persistence with TRANSIENT_LOCAL.
+    DurabilityKind_t durability_red_line = TRANSIENT;
+    if (!isBuiltin)
+    {
+        std::string* persistence_support_transient_local_property = PropertyPolicyHelper::find_property(
+            m_att.properties, "dds.persistence.also-support-transient-local");
+        if (nullptr != persistence_support_transient_local_property &&
+                0 == persistence_support_transient_local_property->compare("true"))
+        {
+            durability_red_line = TRANSIENT_LOCAL;
+        }
+    }
```

Comment:
This duplicated code could go into a private method

Reply:
Done